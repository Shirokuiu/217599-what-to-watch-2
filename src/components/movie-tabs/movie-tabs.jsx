import React, {PureComponent} from "react";
import PropTypes from "prop-types";

import {Tab} from "../../mocks/mocks";

export default class MovieTabs extends PureComponent {
  constructor(props) {
    super(props);

    this.handleTabClick = this.handleTabClick.bind(this);
  }

  handleTabClick(evt) {
    evt.preventDefault();

    const {onTabClick} = this.props;
    const {href, id} = evt.target;

    history.pushState(null, null, href);
    onTabClick(id);
  }

  render() {
    const {movieId, activeTab} = this.props;

    return <nav className="movie-nav movie-card__nav">
      <ul className="movie-nav__list">
        {Tab.moviePage.map((tab, idx) => (
          <li
            key={tab + idx}
            className={
              activeTab.toLowerCase()
                .includes(tab.toLowerCase()) ? `movie-nav__item movie-nav__item--active` : `movie-nav__item`
            }
          >
            <a
              className="movie-nav__link"
              onClick={this.handleTabClick}
              href={`/movie-${movieId}-${tab.toLowerCase()}`}
              id={tab}
            >{tab}</a>
          </li>))}
      </ul>
    </nav>;
  }
}

MovieTabs.propTypes = {
  movieId: PropTypes.number.isRequired,
  onTabClick: PropTypes.func.isRequired,
  activeTab: PropTypes.string.isRequired
};
