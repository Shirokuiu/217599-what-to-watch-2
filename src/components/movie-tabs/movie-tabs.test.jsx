import React from "react";
import renderer from "react-test-renderer";

import {MovieTabs} from "./movie-tabs";

it(`Is movie tabs rendered`, () => {
  const tree = renderer.create(<MovieTabs
    movieId={0}
    onTabClick={jest.fn()}
    activeTab={`details`}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
