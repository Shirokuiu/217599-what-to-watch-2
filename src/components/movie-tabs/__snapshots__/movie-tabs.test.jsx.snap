// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is movie tabs rendered 1`] = `
<nav
  className="movie-nav movie-card__nav"
>
  <ul
    className="movie-nav__list"
  >
    <li
      className="movie-nav__item"
    >
      <a
        className="movie-nav__link"
        href="/movie-0-overview"
        id="Overview"
        onClick={[Function]}
      >
        Overview
      </a>
    </li>
    <li
      className="movie-nav__item movie-nav__item--active"
    >
      <a
        className="movie-nav__link"
        href="/movie-0-details"
        id="Details"
        onClick={[Function]}
      >
        Details
      </a>
    </li>
    <li
      className="movie-nav__item"
    >
      <a
        className="movie-nav__link"
        href="/movie-0-reviews"
        id="Reviews"
        onClick={[Function]}
      >
        Reviews
      </a>
    </li>
  </ul>
</nav>
`;
